src/testing/deploy/upgrade_test.go

```diff
@@ -76,16 +77,48 @@ func TestUpgradeTrigger(t *testing.T) {
 		t.Skip("Skipping upgrade test")
 	}
 	fromVersions := []string{
-		"2.4.6",
-		"2.5.0",
+		"2.7.6",
+		"2.8.5",
 	}
+
+	type ExpectedCommitCount struct {
+		preTrigger1  int
+		preTrigger2  int
+		postTrigger1 int
+		postTrigger2 int
+	}
+
+	getExpectedCommitCountFromVersion := func(version string) ExpectedCommitCount {
+		expectedCommitMap := map[string]ExpectedCommitCount{
```

Comment:
these are the numbers that made my test pass. If I can point to documentation or justify these numbers in a comment, please let me know

Reply:
afaik, they were magic numbers before. now they're magic numbers again, but all in the same place

Reply:
I talked to bryce and he agrees with this.  pre-2.8 there are going to be more commits.